test/e2e-coinjoin-test.py

```diff
@@ -0,0 +1,339 @@
+                           walletclass=walletclass)
+    #the sendpayment bot uses the last wallet in the list
+    wallet_service = wallet_services[end_bot_num - 1]['wallet']
+    jmprint("\n\nTaker wallet seed : " + wallet_services[end_bot_num - 1]['seed'])
+    # for manual audit if necessary, show the maker's wallet seeds
+    # also (note this audit should be automated in future, see
+    # test_full_coinjoin.py in this directory)
+    jmprint("\n\nMaker wallet seeds: ")
+    for i in range(start_bot_num, end_bot_num):
+        jmprint("Maker seed: " + wallet_services[i - 1]['seed'])
+    jmprint("\n")
+    wallet_service.sync_wallet(fast=True)
+    ygclass = YieldGeneratorBasic
+
+    # As per previous note, override non-default command line settings:
+    options = {}
+    for x in ["ordertype", "txfee_contribution", "txfee_contribution_factor",
+              "cjfee_a", "cjfee_r", "cjfee_factor", "minsize", "size_factor"]:
+        options[x] = jm_single().config.get("YIELDGENERATOR", x)
+    ordertype = options["ordertype"]
+    txfee_contribution = int(options["txfee_contribution"])
+    txfee_contribution_factor = float(options["txfee_contribution_factor"])
+    cjfee_factor = float(options["cjfee_factor"])
+    size_factor = float(options["size_factor"])
+    if ordertype == 'reloffer':
+        cjfee_r = options["cjfee_r"]
+        # minimum size is such that you always net profit at least 20%
+        #of the miner fee
+        minsize = max(int(1.2 * txfee_contribution / float(cjfee_r)),
+            int(options["minsize"]))
+        cjfee_a = None
+    elif ordertype == 'absoffer':
+        cjfee_a = int(options["cjfee_a"])
+        minsize = int(options["minsize"])
+        cjfee_r = None
+    else:
+        assert False, "incorrect offertype config for yieldgenerator."
+
+    txtype = wallet_service.get_txtype()
+    if txtype == "p2wpkh":
+        prefix = "sw0"
+    elif txtype == "p2sh-p2wpkh":
+        prefix = "sw"
+    elif txtype == "p2pkh":
+        prefix = ""
+    else:
+        assert False, "Unsupported wallet type for yieldgenerator: " + txtype
+
+    ordertype = prefix + ordertype
+
+    for i in range(start_bot_num, end_bot_num):
+        cfg = [txfee_contribution, cjfee_a, cjfee_r, ordertype, minsize,
+               txfee_contribution_factor, cjfee_factor, size_factor]
+        wallet_service_yg = wallet_services[i - 1]["wallet"]
+
+        wallet_service_yg.startService()
+
+        yg = ygclass(wallet_service_yg, cfg)
+        clientfactory = RegtestJMClientProtocolFactory(yg, proto_type="MAKER")
+        # This ensures that the right rpc/port config is passed into the daemon,
+        # for this specific bot:
+        clientfactory.i = i
+        # This ensures that this bot knows which other bots are directory nodes:
+        clientfactory.set_directory_nodes(directory_node_indices)
+        nodaemon = jm_single().config.getint("DAEMON", "no_daemon")
+        daemon = True if nodaemon == 1 else False
```

Comment:
nit:

```suggestion
        daemon = bool(nodaemon)
```

Reply:
Yes, heh, this is amusingly dumb and a holdover from other code which does essentially the same thing. Will fix here, can't hurt.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182/commits/5fb15ced3608b70d55e40d70ede0f271673fd4ac